packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -92,16 +92,27 @@ export const App = ({
 
   const isSignedIn = isSignedInData?.isSignedIn;
 
-  useMemo(() => {
-    // Don't initialize until integration is ready!
+  const getMetricsEnabled = useCallback(() => {
+    if (metricsLoading || !integration || isSignedIn === undefined) {
+      return;
+    }
+    return data?.account?.metricsEnabled || !isSignedIn;
+  }, [metricsLoading, integration, isSignedIn, data?.account?.metricsEnabled]);
+  const metricsEnabled = getMetricsEnabled();
+
+  useEffect(() => {
```

Comment:
We don't want to use useEffect as it executes at the end of rendering so this won't execute until all of the children finish rendering but we might want Glean to emit events when the children render.

Reply:
@chenba I noticed using the `useMemo` it initalizes Glean several times, at least with console logging.

<img width="685" alt="Screenshot 2024-04-11 at 2 12 08 PM" src="https://github.com/mozilla/fxa/assets/1295288/8284bec3-1fe4-428f-b6df-45333cae030d">

Are there certain events that need to be initalized before rendering?

Reply:
> useMemo it initalizes Glean several times

Maybe dependencies for useMemo changed?  I don't think it's harmful to init glean.js more than once though.

> Are there certain events that need to be initalized before rendering?

Most if not all of the `.view` events are called in a `useEffect` of their component.